Sources/ConcordiumSwiftSdk/NodeClient.swift

```diff
@@ -3,11 +3,23 @@ import GRPC
 import NIOCore
 import NIOPosix
 
-public class Client {
-    let grpc: Concordium_V2_QueriesNIOClient
+public protocol NodeClient {
+    func getCryptographicParameters(at block: BlockIdentifier) async throws -> CryptographicParameters
 
-    public init(channel: GRPCChannel) {
-        grpc = Concordium_V2_QueriesNIOClient(channel: channel)
+    func getNextAccountSequenceNumber(of address: AccountAddress) async throws -> NextAccountSequenceNumber
+
+    func getAccountInfo(of account: AccountIdentifier, at block: BlockIdentifier) async throws -> AccountInfo
+}
+
+public class GrpcNodeClient: NodeClient {
+    let grpc: Concordium_V2_QueriesClientProtocol
+
+    init(_ grpc: Concordium_V2_QueriesClientProtocol) {
+        self.grpc = grpc
+    }
+
+    public convenience init(channel: GRPCChannel) {
```

Comment:
We should probably wrap the GRPCChannel as well, so users of this client implementation only has to import the ConcordiumSwiftSDK and not GRPC and NIOPosix as well.

Reply:
I like the idea but it would require the client to "own" the channel in order to manage its lifecycle (or add a "manager" wrapper of the client). See the client tests or example CLI which add deferred closing of the group/channel.

The way it is now is similar to how it's done in Rust SDK (see [usage](https://github.com/Concordium/concordium-rust-sdk/#basic-usage), note that `v2::Endpoint` is a direct re-export of `tonic::Endpoint`).

The Java SDK [adds some wrappers](https://github.com/Concordium/concordium-java-sdk/tree/main#example-of-instantiating-the-client) around the gRPC types, but they use `ManagedChannel` which I think is not in the Swift lib and therefore something we'd have to write outselves. And frankly I find the Java solution to be overly complicated with very little gain. I'd prefer just having the user initiate the channel themselves. That will allow them to rely on official channels without having to untangle our abstractions.